pgcontents/tests/test_pgmanager.py

```diff
@@ -185,6 +185,50 @@ def test_get_file_id(self):
         cm.rename(path, updated_path)
         self.assertEqual(id_, cm.get_file_id(updated_path))
 
+    def test_rename_file(self):
+        cm = self.contents_manager
+        nb, nb_name, nb_path = self.new_notebook()
+        nb_model = cm.get(nb_path)
+        assert nb_model['name'] == 'Untitled.ipynb'
+
+        # A simple rename of the file within the same directory.
+        file_id = cm.get_file_id('Untitled.ipynb')
+        renamed = cm.rename_file(nb_path, 'new_name.ipynb')
```

Comment:
I noticed that but actually thought calling `rename_file` was better/more explicit considering it's the function we are overriding. I can change this to `rename`, but if that's always the preferred method I kind of wish `rename_file` was made private 😕

Reply:
In a language like C++ or Java, `rename_file` would probably have been "protected" (i.e., visible to subclasses, but not to code outside the class hierarchy); it's publicly-named on the base class because it's designed for subclasses to override, but it's not really public for external consumers.